### PR TITLE
[ADD] purchase_order_sale_order_nrq

### PR DESCRIPTION
**Pull request by TimLai125**
- Add `sale_ids` to Purchase Order and Account Invoice.
- Link Sales Order's `project_id` with Purchase Order Line's `account_analytic_id`
- Link Sales Order to Account Invoice by checking the `purchase_id` in the `invoice_line_ids` of the Account Invoice.
- Add filters to list views of Purchase Order and Account Invoice.

### CODE REVIEW

File: purchase_order_sale_order_nrq/models/account_invoice.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        string='Related Sales Order(s)'
+    )
+
+    @api.multi
+    @api.depends('invoice_line_ids')
+    def _compute_sale_ids(self):
+        for invoice in self:
+            sale_ids = []
+            for invoice_line in invoice.invoice_line_ids:
+                if invoice_line.purchase_id:
```
Comment (yostashiro) [inline purchase_order_sale_order_nrq/models/account_invoice.py]: We'd better not depend on `purchase_id` here.  There can be cases of a vendor bill not created from a purchase order.  Better apply the same logic as purchase order upon adding the consideration on invoice type (apply the logic only when `type in ['in_invoice', 'out_invoice']`).

**Review by yostashiro — COMMENTED** [review]
I think the current design is basically fine.  After some thoughts, however, I came to think we might want to restructure the design to keep `sale_id` in the line level (i.e. `purchase.order.line` and `account.invoice.line`) to simplify the logic (the field should be a stored computed field depending on `analytic_account_id`) and avoid computation overhead (although it is a minor concern).

Then, to facilitate the filter requirement, we can do the same as `product_id` field in SO https://github.com/odoo/odoo/blob/ef48b81ab127dd717a4a9d106edd5c51121bc8f7/addons/sale/models/sale.py#L159 - we can add `sale_id` (a related field to line `sale_id`) in `purchase.order` and `account.invoice`.

I think this will make the code slightly more maintainable.  What do you think?

File: purchase_order_sale_order_nrq/data/ir_action_data.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="purchase_order_sale_ids_initialize" model="ir.actions.server">
```
Comment (yostashiro) [inline purchase_order_sale_order_nrq/data/ir_action_data.xml]: I suppose we do not need these records for updating stored computed fields.  Let me know if this is not the case.

Comment (TimLai125) [reply purchase_order_sale_order_nrq/data/ir_action_data.xml]: Yes, you are correct. It behaves strangely on my development environment that the invoice side's fields are not initialized. Let me remove this part for now

File: purchase_order_sale_order_nrq/models/account_invoice.py (hunk 1 above)
Comment (TimLai125) [reply purchase_order_sale_order_nrq/models/account_invoice.py]: Yes, it sounds better to keep the field in the line level. I will make an update to this PR later

**Pull request merged**